src/vs/workbench/api/common/extHostDocumentsAndEditors.ts

```diff
@@ -97,6 +97,13 @@ export class ExtHostDocumentsAndEditors implements ExtHostDocumentsAndEditorsSha
 					}
 				}
 				if (!ref) {
+					// {{SQL CARBON EDIT}}
+					// Add URI of the notebook that is using this document for a cell's editor.
+					if (!data.notebook && data.notebookUri) {
```

Comment:
We can't re-use this existing notebook field because it only exists in the extension host after converting the IModelAddedData created from the original ITextModel into this IExtHostModelAddedData.